common/util/BinaryWriter.h

```diff
@@ -4,12 +4,13 @@
  * @file BinaryWriter.h
  * Write raw data like a stream.
  */
-
+#include <unistd.h>
```

Comment:
`unistd` will have to be mac/linux only, it's not on windows. I don't think there's anything in here that needs `unistd.h`, maybe you could add `#include <cstdio>` instead? (I'm assuming the issue was with `fopen`/`fwrite`/`fclose`, which it's better to use `cstdio` because that works on mac/windows/linux)

Reply:
It looks like there are some other cases where `unistd.h` was implemented over `cstdio`  that I don't remember implementing, `common/cross_sockets/xsocket.cpp#L11` for example. Are there certain cases where we need `unistd.h` specifically or would it be safe to use cstdio to replace everything?

Reply:
There are some things that are only in `unistd.h`, like `usleep` or `close`, so we do have to use it some cases.  In `xsocket` we need it for `close`, which closes a unix file descriptor (for the unix sockets)